### ISSUE TITLE
Backport nav2_graceful_controller to Iron

### ISSUE THREAD

**Issue by gennartan** [ros-navigation/navigation2#4658]
## Feature request

Backport nav2_graceful_controller to Iron. 

#### Feature description

This new controller is required by the opennav_docking project [1]. As I am tyring to integrate a docking behavior to my robot, it would be nice to be able to use opennav docking framework.

I don't know if there is a scheduled backport that will already integrate the new controller, so I am creating this feature request first. If this is not the case, then I can look into it to add this feature by cherry-picking the required commits for Iron.

[1] https://github.com/open-navigation/opennav_docking

**Comment by SteveMacenski** [comment]
Iron is EOL in only 8 weeks, so we made the call to not backport and support Nav2 Docking in Iron. If you'd like to backport it to Iron, I'd be happy to review / merge the PR. 

The only caveats is that if there are any API breaking changes between Iron and Jazzy/Rolling, you might need to make minor tweaks to the code. Just note in your PR what changes beyond copy/pasting you made to have it compile and be API compliant with the `nav2_core` Planner plugin API.  

Noting those changes explicitly will greatly accelerate my review to only look at what was changed and otherwise rubberstamp and merge within the day.
